crates/futures/src/lib.rs

```diff
@@ -289,10 +296,68 @@ fn _future_to_promise(future: Box<Future<Item = JsValue, Error = JsValue>>) -> P
         Waiting(Arc<Package>),
     }
 
-    // No shared memory right now, wasm is single threaded, no need to worry
-    // about this!
-    unsafe impl Send for Package {}
-    unsafe impl Sync for Package {}
+    struct Waker {
+        buffer: SharedArrayBuffer,
```

Comment:
This I don't think will store `SharedArrayBuffer` but rather use `wasm_bindgen::memory().buffer()` as the `SharedArrayBuffer` accessor

Reply:
oh, fine! but how do I allocate it?